hbase-common/src/main/java/org/apache/hadoop/hbase/nio/RefCnt.java

```diff
@@ -60,4 +64,30 @@ protected final void deallocate() {
   public final ReferenceCounted touch(Object hint) {
     throw new UnsupportedOperationException();
   }
+
+  public void setInnerRefCnt(Optional<RefCnt> refCnt) {
```

Comment:
The RefCnt is a general reference count class here... The refCnt with  interRefCnt is mainly used for BucketEntry ?  Could  we create a subclass for BuckentEntry only ?  I mean the general ByteBuff use the customized will be resource-consuming ...